tool/src/java/org/sakaiproject/gradebookng/tool/model/GradebookUiSettings.java

```diff
@@ -49,7 +50,8 @@
 	public GradebookUiSettings() {
 		//defaults. Note there is no default for assignmentSortOrder as that requires an assignmentId which will differ between gradebooks
 		this.categoriesEnabled = false;
-		this.assignmentVisibility = new HashMap<Long, Boolean>();
+		this.assignmentVisibility = new ConcurrentHashMap<Long, Boolean>();
+		this.categoryScoreVisiblity = new ConcurrentHashMap<String, Boolean>();
 		this.nameSortOrder = GbStudentNameSortOrder.LAST_NAME;
```

Comment:
Hm.. Is there a potential concurrency issue here?  I started seeing some weird behaviour when I hit "Hide All" as it fires off an ajax post for all the grade items to update the same Hash.

I'm wondering if we need to make access or setting of this session object a little more robust to concurrent requests?